File: solutions/0905. Sort Array By Parity/0905_1.cpp
// Time complexity: O(n)
// Space complexity: O(n)
class Solution {
public:
	vector<int> sortArrayByParity(vector<int>& nums) {
		vector<int> ans;
		ans.reserve(nums.size());
        for (int num : nums) {
            if ((num & 1) == 0) {
                ans.push_back(num);
            }
        }
        for (int num : nums) {
            if ((num & 1) != 0) {
                ans.push_back(num);
            }
        }
        assert(ans.size() == nums.size());
        return ans;
	}
};

File: solutions/0905. Sort Array By Parity/0905_1.js
// Time complexity: O(n)
// Space complexity: O(n)
var sortArrayByParity = function(nums) {
    const ans = [];
    for (const num of nums) {
        if ((num & 1) === 0) {
            ans.push(num);
        }
    }
    for (const num of nums) {
        if ((num & 1) !== 0) {
            ans.push(num);
        }
    }
    return ans;
};